storage/src/range_set.rs

```diff
@@ -0,0 +1,598 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
```

Comment:
This needs a top-level comment saying where this originally came from and an overview of the modifications we made from it. It's possible we can't relicense it here.

Reply:
If it's not derived from anything, then great! It will need a more thorough review on my next pass though.